In [2]:
import pandas as pd
import numpy as np

# Print “Hello, Welcome to Python and Data Analytics Project!” 
print("Hello, Welcome to Python and Data Analytics Project!")
print("I am Chandrakanth Bohini, Email:chanduapps123@gmail.com, Mobile: +91-8179843111")

Hello, Welcome to Python and Data Analytics Project!
I am Chandrakanth Bohini, Email:chanduapps123@gmail.com, Mobile: +91-8179843111


In [4]:
# Reading Data using PANDAS
#Used pandas read_csv() function to read the required file 

pdap = pd.DataFrame(pd.read_csv('/content/train.csv'))
#Used pandas dataframe() function to convert file into data frame.
pdap.head()
# pandas shape is used to know the numbers of records in the csv file-- Rows and columns
df= pdap.shape
print("Number of rows and columns in the csv file are : ",df)
dfrows= pdap.shape[0]
print("Number of rows in the csv file are : ",dfrows)
dfcolumns= pdap.shape[1]
print("Number of columns in the csv file are : ",dfcolumns)

Number of rows and columns in the csv file are :  (891, 12)
Number of rows in the csv file are :  891
Number of columns in the csv file are :  12


### Discription of the attributes of the dataset
#### Pclass : Passanger Class(1=1st class, 2=2nd class, 3=3rd class)
#### Survived : Survival(0= NO(Not survied or died),  1= YES(Survied or lived))
#### Name : Passanger Name
#### Sex : Passanger Gender
#### Age : Passanger Age
#### SibSp : Passanger's siblings or spouses aboard
#### Parch : Passanger's Parents or Children aboard
#### Ticket : Passanger Ticket Number
#### Fare : Passanger Ticket Fare
#### Cabin : Passanger's Cabin
#### Cabin : Port of Embarkation(C= Cherbourg, Q=Queenstown, S= Southampton)



# Handling Null Values

#### The dataset may contain missing values, we cannot ignore those missing values.
#### We can Either drop entire row or column or Fill the missing values wih some approx values.
#### Example Mean of all values for that column.

In [5]:
null_value_eachcolum = pdap.isnull().sum() # This shows NULL Values in each column
print("Null Values in each columns are as follows")
print(1*"\n")
null_value_eachcolum

Null Values in each columns are as follows




PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

### Seperating out the columns which have more than 35% of the values missing in the dataset 


###### Out of 891 rows for Age column there are 177 rows and for Cabin there are 687 rows and for Embarked there are 2 having null values 
###### We are storing  only those  columns  which have more than 35% of the data missing.
#### As we know we can handling these NULL Values by dropping entire rows r columns or by filling Mean of all values for that column

In [6]:
# Here "df" is called from the top.
# where we have declared "df= pdap.shape" is the pandas shape, 
# It is used to know the numbers of records in the csv file-In the tuple[0] indicates the first value-"dfrows"--891
# "null_value_eachcolum" is shows NULL Values in each column 
drop_col = null_value_eachcolum[null_value_eachcolum > (35/100 *dfrows)]
drop_col
print("Column having more than 35% of data as null values are  :", drop_col)
# This command or code will print the column name along with records 
# which satisfify the above condition. Here Cabin is having more than 35% of data as null values.

Column having more than 35% of data as null values are  : Cabin    687
dtype: int64


### Note: There is no specific number  after which  you should drop the column.

In [7]:
# Dropped the Cabin 
pdap.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [8]:
drop_col.index # index is having the column Cabin

 ##pandas.DataFrame.drop
##Drop specified labels from rows or columns.

##Remove rows or columns by specifying label names and corresponding axis, or
##by specifying directly index or column names. 
##When using a multi-index, labels on different levels can be removed by specifying the level


#Parameters
#labels : single label or list-like
#Index or column labels to drop.

#axis:  {0 or ‘index’, 1 or ‘columns’}, default 0
#Whether to drop labels from the index (0 or ‘index’) or columns (1 or ‘columns’).

#index:  single label or list-like
#Alternative to specifying axis (labels, axis=0 is equivalent to index=labels).

#columns: single label or list-like
#Alternative to specifying axis (labels, axis=1 is equivalent to columns=labels).

#level:  int or level name, optional
#For MultiIndex, level from which the labels will be removed.

#inplace: bool, default False
#If False, return a copy. Otherwise, do operation inplace and return None.

#errors: {‘ignore’, ‘raise’}, default ‘raise’
#If ‘ignore’, suppress error and only existing labels are dropped.

#Returns: DataFrame or None
#DataFrame without the removed index or column labels or None if inplace=True.

#Raises: KeyError
#If any of the labels is not found in the selected axis.

Index(['Cabin'], dtype='object')

In [9]:
#To delete the column from dataframe
pdap.drop(drop_col.index, axis=1, inplace=True)
pdap.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Embarked         2
dtype: int64

Note : We do not drop all other columns, We have dropped Cabin column since it was not much useful

## Other columns can be filled with mean of all values for that column
## Pandas.fillna can be used here

In [11]:
Mean_fill_value = pdap.fillna(pdap.mean(), inplace=True) # It fill the dataframe with specific value.

# Example for Age value it will calculate the Mean of the rest of the vlaues and all the NULL Values 
# are filled with the mean value. Hear we have taken Mean because the Age column is important column and 
# can not be a Null values, the appropiate value should be the Mean Value.



In [12]:
print("Null Values in each columns  :",pdap.isnull().sum())

Null Values in each columns  : PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       2
dtype: int64


### Note : Here Age has been filled by the mean value but the column "Embarked" is not filled with mean values 
### because this columns has strings and we cannot calcualte the mean of strings.

In [13]:
# To dispay a column details 
Col_details= pdap['Embarked'].describe()
print("Dispay a column details  :",Col_details)

Dispay a column details  : count     889
unique      3
top         S
freq      644
Name: Embarked, dtype: object


In [14]:
# Hear count is showing the Not Null values (Out of 891 records 889 are not null and 2 are null values)
# unique will shows the same values in the column, Example C,Q,S
# Top shows the higest unique values. Example - "S"
# freq tells the count of the Top--How many records are there
# Now we will fill the NULL values with the most frequent value in the column.

pdap['Embarked'].fillna('S',inplace=True) # Filling the Null values with "S" inthe Embarked Column

print("Null Values in each columns  :",pdap.isnull().sum())


pdap.corr() # This tells about the co relation between different columns.

Null Values in each columns  : PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64


PassengerId  Survived    Pclass  ...     SibSp     Parch      Fare
PassengerId     1.000000 -0.005007 -0.035144  ... -0.057527 -0.001652  0.012658
Survived       -0.005007  1.000000 -0.338481  ... -0.035322  0.081629  0.257307
Pclass         -0.035144 -0.338481  1.000000  ...  0.083081  0.018443 -0.549500
Age             0.033207 -0.069809 -0.331339  ... -0.232625 -0.179191  0.091566
SibSp          -0.057527 -0.035322  0.083081  ...  1.000000  0.414838  0.159651
Parch          -0.001652  0.081629  0.018443  ...  0.414838  1.000000  0.216225
Fare            0.012658  0.257307 -0.549500  ...  0.159651  0.216225  1.000000

[7 rows x 7 columns]

# Correlation: 
### Correlation summarizes the strength and direction of the linear (straight-line) association between two quantitative variables. Denoted by r, it takes values between -1 and +1. A positive value for r indicates a positive association, and a negative value for r indicates a negative association

### We can observed there is a good correlation b/w Pclass and Fare

In [15]:
# We can make a new column family_size by combining the below columns
# SibSp : Passanger's siblings or spouses aboard
# Parch : Passanger's Parents or Children aboard

pdap['FamilySize'] = pdap['SibSp'] + pdap['Parch'] # Added new column by combining SibSp and Parch columns
pdap.drop(['SibSp', 'Parch'], axis=1, inplace=True)# Dropping the SibSp and Parch Columns 
pdap.corr() # Correlation b/w Columns

PassengerId  Survived    Pclass       Age      Fare  FamilySize
PassengerId     1.000000 -0.005007 -0.035144  0.033207  0.012658   -0.040143
Survived       -0.005007  1.000000 -0.338481 -0.069809  0.257307    0.016639
Pclass         -0.035144 -0.338481  1.000000 -0.331339 -0.549500    0.065997
Age             0.033207 -0.069809 -0.331339  1.000000  0.091566   -0.248512
Fare            0.012658  0.257307 -0.549500  0.091566  1.000000    0.217138
FamilySize     -0.040143  0.016639  0.065997 -0.248512  0.217138    1.000000

### FamilySize is not have much correlation with Survived rate.
### Pclass is ahving good correlation with Survived rate.
### Checking weather the person was alone or not can effect the survived rate.

In [16]:
pdap['Alone'] = [0 if pdap['FamilySize'][i]>0 else 1 for i in pdap.index] # Creating a new column "Alone" 
# This Column will print the value "0" if the familysize is greater than 0 and else "1"
pdap.head()

PassengerId  Survived  Pclass  ... Embarked FamilySize  Alone
0            1         0       3  ...        S          1      0
1            2         1       1  ...        C          1      0
2            3         1       3  ...        S          0      1
3            4         1       1  ...        S          1      0
4            5         0       3  ...        S          0      1

[5 rows x 11 columns]

In [17]:
# Calculating the Mean values for Alone and Survived
pdap.groupby(['Alone'])['Survived'].mean()

# Groupby for the "Alone" Column the Value of 0 and 1 it will calculate the mean

Alone
0    0.505650
1    0.303538
Name: Survived, dtype: float64

#### This means that there is greater chance for survival who are not alone.
#### If he had a family the better chance of survival.-- This because might he/she belonging to rich class and might be prioritized over other.

In [18]:
pdap[['Alone','Fare']].corr() # Correlation b/w Alone and fare

Alone      Fare
Alone  1.000000 -0.271832
Fare  -0.271832  1.000000

In [19]:
# Here we can see if the person  was not alone, the chance, the ticket price is higher

pdap['Sex'] = [0 if pdap['Sex'][i]=='male'else 1 for i in pdap.index] # 1 for female and 0 for male
pdap.groupby(['Sex'])['Survived'].mean()

## Female has the more chance of suvival over Male

Sex
0    0.188908
1    0.742038
Name: Survived, dtype: float64

In [20]:
#Embarked with the Survived

pdap.groupby(['Embarked'])['Survived'].mean()

# The Survival is more who has embarked as "C"

Embarked
C    0.553571
Q    0.389610
S    0.339009
Name: Survived, dtype: float64

# Conclusion:

   - Female passengers are more prioritized Over Men.
   - Pclass is having high survival rate than others.
   - Who are travelling with family having high suvival rate than who are travellig Alone.
   - Passengers who boarded/Embarked at "C- Cherbourg" survived more than others.